CMakeLists.txt

```diff
@@ -42,14 +43,23 @@ endif()
 if(BUILD_FUZZ)
     message(STATUS "Build for fuzzing")
     add_compile_options("-DFUZZ")
-endif()
-if(FUZZ_TARGET STREQUAL "semantic")
-    message(STATUS "Fuzzing seantic analyzer")
-    add_compile_options("-DTEST_SEMANTIC")
-elseif(FUZZ_TARGET STREQUAL "codegen")
-    message(STATUS "Fuzzing codegen")
-else()
-    message(FATAL_ERROR "Unsupported FUZZ_TARGET")
+
+    if(FUZZ_TARGET STREQUAL "semantic")
+        message(STATUS "Fuzzing seantic analyzer")
+        add_compile_options("-DTEST_SEMANTIC")
```

Comment:
I think `target_compile_options()` is better b/c it doesn't proliferate the macro everywhere. Also probably more portable.

Reply:
I didn't know that option. Now I move to fuzz related settings to src/CMakeList.txt and use it.

Reply:
ah, target_compile_definitions looks better.

Reply:
hmm.. I tried to use target_compile_definitions but it does not work the same as the target_compile_options (encountered linking error with libfuzzer.a). CMake is difficult... I'll keep the current code.